### PR TITLE
Refactor(eos_cli_config_gen)!: Require `enabled: true` under `vlan_interfaces.[].ip_attached_host_route_export`

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Require `enabled: true` under `vlan_interfaces.[].ip_attached_host_route_export`

## Related Issue(s)

Part of #2288 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

To avoid ambiguous YAML data input, the data model for `vlan_interfaces.[].ip_attached_host_route_export` has been updated to require
an `enabled: true` key to be added.

Example with old data model (shown with old dictionary form input):

```yaml
vlan_interfaces:
  Vlan86:
    ip_address: 10.10.83.1/24
    ip_attached_host_route_export: {}
```

Same example with new data model (shown with new list form input):

```yaml
vlan_interfaces:
  - name: Vlan86
    ip_address: 10.10.83.1/24
    ip_attached_host_route_export:
      enabled: true
```

The change has been incorporated into `eos_designs` so action is only required when defining `structured_configuration` directly.

## How to test

- Updated molecule for eos_cli_config_gen. No config change.
- Reran eos_designs after updating logic. No config change. Expected change in structured.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/intended/configs/DC2-SUPER-SPINE1.cfg [hunk 1]
```diff
@@ -123,6 +123,7 @@ router bgp 65200
    neighbor 11.1.2.0 peer group IPv4-UNDERLAY-PEERS
    neighbor 11.1.2.0 remote-as 65100
    neighbor 11.1.2.0 description DC1-SUPER-SPINE1
+   no neighbor 11.1.2.0 bfd
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/intended/configs/DC2-SUPER-SPINE1.cfg]: This looks surprising that it pops up here - @carlbuchmann did we merge a non passing PR when we modified bfd?

**Review by gmuloc — APPROVED** [review]
LGTM - the comment is only relative to changes not related to this PR

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/intended/configs/DC2-SUPER-SPINE1.cfg (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/molecule/eos_designs-twodc-5stage-clos/intended/configs/DC2-SUPER-SPINE1.cfg]: Fixed in #2778

**Pull request merged**